### PR TITLE
[ENH] FreeViz: new widget

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
No FreeViz widget.

##### Description of changes
1) Icon needed (Issue #2519 ). Maybe we can use Linear Projection's icon.
2) Tests coverage so far: 82% (Scatter Plot has 74%) Missing GUI tests!
3) Scripting part in a separate PR (need to discuss with somebody).
4) Now can handle data sets with many attributes.
5) Highly recommended that PR #2541 is accepted and merged before this PR. DONE

##### Includes
- [x] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/visualize/owfreeviz.py [hunk 1]
```diff
@@ -0,0 +1,925 @@
...
+    func = lambda x, trigo: np.fromfunction(
+        lambda i: x + CIRCLE_RADIUS * trigo(2. * np.pi * i / 120.), (121,),
+        dtype=int)
+    circle_x = func(x, np.cos)
+    circle_y = func(y, np.sin)
+    color = QColor(0, 0, 255)  # blue
+    circle = pg.PlotCurveItem(
+        x=circle_x, y=circle_y,
+        pen=pg.mkPen(color, width=3),
+        antialias=False
```
Comment (janezd) [inline Orange/widgets/visualize/owfreeviz.py]: `antiAlias=True`

Comment (janezd) [reply Orange/widgets/visualize/owfreeviz.py]: However, I think that this looks better and more informative.

```
class MoveIndicator(pg.GraphicsObject):
    def __init__(self, x, y, parent=None, line=QLineF(), text="", **kwargs):
        super().__init__(parent, **kwargs)
        self.arrows = [
            pg.ArrowItem(pos=(x - 0.07 * cos(radians(angle)),
                              y + 0.07 * sin(radians(angle))),
                         parent=self, angle=angle,
                         headLen=13, tipAngle=45,
                         brush=pg.mkColor(128, 128, 128))
            for angle in (0, 90, 180, 270)]

    def paint(self, painter, option, widget):
        pass

    def boundingRect(self):
        return QRectF()


def add_circle(x, y):
    return MoveIndicator(x, y)
```

Of course, eliminate `add_circle` and use `MoveIndicator` directly in the code.

**Comment by janezd** [comment]
If you mark a selection (that is, click in the graph and start dragging) and you approach the arrow, the move indicator (currently a blue circle) appears. These shouldn't appear while selecting.

**Comment by jerneju** [comment]
@janezd: Thank you for the comments. Scripting part (#2563) is ok?

File: Orange/widgets/visualize/utils/plotutils.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+class TextItem(pg.TextItem):
+    if not hasattr(pg.TextItem, "setAnchor"):
+        # Compatibility with pyqtgraph <= 0.9.10; in (as of yet unreleased)
+        # 0.9.11 the TextItem has a `setAnchor`, but not `updateText`
+        def setAnchor(self, anchor):
+            self.anchor = pg.Point(anchor)
+            self.updateText()
+
+
+class AxisItem(pg.GraphicsObject):
```
[the remark this answers is not in the document]
Comment (jerneju) [reply Orange/widgets/visualize/utils/plotutils.py]: No, this class is not related to `pg.AxisItem`. Renamed.

File: Orange/widgets/visualize/owfreeviz.py (hunk 1 above)
Comment (jerneju) [reply Orange/widgets/visualize/owfreeviz.py]: Yes, that looks better.

**Comment by janezd** [comment]
If we merge this, the icon I proposed can be used until @larazupan provides a better one.

**Comment by jerneju** [comment]
@janezd : rebased.

**Pull request merged**